### PR TITLE
Nadam support the flag for "maximize"

### LINKED ISSUE #126642: [optim] add missing 'maximize' parameter to LBFGS, NAdam and RAdam optimizers
### 🚀 The feature, motivation and pitch

Currently, ```LBFGS```, ```NAdam``` and ```RAdam``` optimizers do not have the ```maximize``` parameter in their constructors, which means they cannot be used with some loss functions (like R2Score).

### Alternatives

In the case ```maximize``` can't be implemented due to a constraint of the optimizer, it would be nice to make it clear that ```maximize``` is intentionally not supported. Some possible suggestions:
* Update the documentation to mention the reason maximize isn't supported for the optimizer
* Still support the parameter but throw an error when maximize=True

### Additional context

_No response_

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### PR DESCRIPTION
**Pull request by huihoaan**
Fixes https://github.com/pytorch/pytorch/issues/126642

### CODE REVIEW

File: torch/optim/nadam.py [hunk 1]
```diff
@@ -276,12 +285,13 @@ def _single_tensor_nadam(
...
     eps: float,
     decoupled_weight_decay: bool,
+    maximize: bool,
     capturable: bool,
     differentiable: bool,
     has_complex: bool,
 ):
     for i, param in enumerate(params):
-        grad = grads[i]
+        grad = grads[i] if not maximize else -grads[i]
```
Comment (vadimkantorov) [inline torch/optim/nadam.py:294]: this leads to materialization copy of grads, right?

**Review by janeyx99 — APPROVED** [review]
nice! can you also delete line 644 in common_optimizers.py?

File: torch/optim/nadam.py (hunk 1 above)
Comment (janeyx99) [reply torch/optim/nadam.py:294]: do you mean it’ll make a copy to compute the negation instead of just referencing?

if so, i think so, but it’s ok

Comment (vadimkantorov) [reply torch/optim/nadam.py:294]: Yeah, maybe the negative bit functionality can be cool here :) Originally I was thinking that it could negate inplace... but maybe it's not a very widely used feature anyway...

**Comment by huihoaan** [comment]
> nice! can you also delete line 644 in common_optimizers.py?

Yes, this comment should be deleted

File: torch/optim/nadam.py (hunk 1 above)
Comment (janeyx99) [reply torch/optim/nadam.py:294]: we don’t wanna modify the original grad in place!

**Comment by janeyx99** [comment]
there are also lint failures on the previous commit, please run lintrunner locally before committing.

**Comment by huihoaan** [comment]
> there are also lint failures on the previous commit, please run lintrunner locally before committing.

lint failures on the previous commit is my changed file, I have succeed in local.

**Review by janeyx99 — APPROVED** [review]
